src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -425,10 +425,19 @@ internal static string GetRepositoryPath(IMSBuildProject project, ISettings sett
         /// </summary>
         /// <param name="project">The <see cref="IMSBuildItem" /> representing the project.</param>
         /// <returns>The full path to the restore output directory for the specified project.</returns>
-        internal static string GetRestoreOutputPath(IMSBuildProject project)
+        internal string GetRestoreOutputPath(IMSBuildProject project)
         {
             string outputPath = project.GetProperty("RestoreOutputPath") ?? project.GetProperty("MSBuildProjectExtensionsPath");
 
+            if (string.IsNullOrWhiteSpace(outputPath))
+            {
+                // If the project doesn't supply a restore output path, it doesn't support restore.  The properties above are defined in the .NET SDK and if a project does not have them set then the project
+                // author isn't importing something from the SDK which is not supported.
+                MSBuildLogger.LogVerbose($"The project '{project.FullPath}' does not specify a value for the properties 'RestoreOutputPath' or 'MSBuildProjectExtensionsPath' and will not be restored.");
```

Comment:
What is everyone's thoughts on logging a warning vs a low importance message?  It might be a project they just don't care about so I don't want to create too much noise.  But I don't want people pulling their hair out trying to figure out why the project isn't being restored.

Reply:
Personally, I think a warning makes sense here.

This is usually a custom project, so unlikely that users hit this frequently. Given that these projects implement the NuGet targets, I think it's fine to warn as long as the warnings are suppressible. (which given where the warning is raised right now, I doubt they would be).

Reply:
Given that *it might be* super difficult to have a suppressible warning, maybe just a simple message is better though. 
The ROI might not be there. 

A side question, how does regular restore behave?

Reply:
Good question, it took me a while to figure it out as its hard to get a project in a state where restore works but these properties aren't set.  Here's what I did:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
    <RestoreOutputPath></RestoreOutputPath>
    <MSBuildProjectExtensionsPath></MSBuildProjectExtensionsPath>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
  </ItemGroup>
</Project>
```

This produces this error: 
```
 C:\Program Files (x86)\Visual Studio 2019 Preview\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.targets(130,5): error : Invalid restore input. Missing required property 'OutputPath' for project type 'PackageReference'. Input files: D:\Stuff\no-restoreoutputpath\ProjectA.proj. [D:\Stuff\no-restoreoutputpath\ProjectA.proj]
```

This error is logged by [ValidateProjectSpecPackageReference](http://index/?leftProject=NuGet.Commands&leftSymbol=o8r2knf7kuq7).  Maybe this fix should be to just let null flow into the package spec and let this error happen during validation?

Reply:
yeah, that's probably easiest. It also has the benefit of minimizing duplication :)

Reply:
Okay I've updated it to just return `null` and let the package spec validation handle the error in a uniform way.

Reply:
Sounds good. I approved. Can you please add a test for missing output path?

Reply:
Done!